modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -349,7 +349,10 @@ def serialize_dtype(self, dtype):
             Serialized data type.
         """
         _warn_if_unsigned(dtype)
-        return {"type": self._DTYPE_STRINGS[dtype.name], "nullable": True}
+        try:
+            return {"type": self._DTYPE_STRINGS[dtype.name], "nullable": True}
+        except KeyError:
+            raise TypeError(f"Unsupported dtype: {dtype}")
```

Comment:
where do we catch this exception?

Reply:
It's not catched, but there were 2 tests failing with the following error:
AssertionError: Got Modin Exception type <class 'KeyError'>, but pandas Exception type <class 'TypeError'> was expected